### PR TITLE
Updated SiriusXM Connector with more filters

### PR DESCRIPTION
**Pull request by jbwharris**
Related to #2222 #3319 

Added some improved filtering for station IDs and DJ handles. I did some testing and playing around with finding different criteria for exclusion and found some weird edge cases. 

- My filter now checks the artist and track instead of just artist, as it increases the percentage of properly filtering the station IDs and Twitter handles
- I discovered that filtering by ﹫ tends to be too broad. The character is used sometimes with certain channels to denote the artist, most prominently The Grateful Dead channel using ﹫greatfuldead all the time. Or it can be used in some like "Live ﹫ SiriusXM" which would still be a valid scrobble.
- I added a few Twitter account handles, but found the list tended to be too extensive to filter against them all. We could compile such a list, but I don't readily have that list handy. The good thing is that some of the filters in tandem with the Twitter filters tends to get most of the cases (ie. like where they have a host handle with a URL or hashtag)
- I added a ".c" filter which would catch the vast majority of URLs, both .ca and .com. I'm not sure if there are other regionalizations of SiriusXM that might have URLs that should be filtered. I didn't encounter any others.
- Adding #️⃣ tends to catch a lot of the ones we want to filter. I did note this had a possibility to excluding classical tracks, but looking at the lastfm data I didn't see much indication people were probably scrobbling those channels.
- I tried having "siriusxm" but found that one was way too broad. It would exclude live sessions or some show names. Didn't want to limit how some people might use Web Scrobbler for tracking.
- I also tweaked the trackArtSelector, as it had been pulling null previously

**Scrobbling Data I used to find filter criteria**
- https://www.last.fm/search/tracks?q=%40sxm
- https://www.last.fm/search/tracks?q=siriusxm
- https://www.last.fm/search/tracks?q=fb.com

### CODE REVIEW

File: src/connectors/siriusxm-player.ts [hunk 1]
```diff
@@ -10,34 +10,47 @@ export {};
...
+	const filteredTerms = [
+		"@siriusxm",
+		"@jennylsq",
+		"@radiomadison",
+		"@morningmashup",
+		"josiah",
+		"1-877-33-sirius",
+		"@sxm", //will broadly catch a bunch of sxm Twitter handles
+		"altnation",
+		".c", // will catch .com and .ca URLs
```
Comment (inverse) [inline src/connectors/siriusxm-player.ts:40]: Would it be better to be more explicit and have two entries for `.ca` and `.com`?

Comment (jbwharris) [reply src/connectors/siriusxm-player.ts:40]: @inverse I can do that. Probably getting too cute doing that anyways 😂

Comment (jbwharris) [reply src/connectors/siriusxm-player.ts]: @inverse I pushed that change over

File: src/connectors/siriusxm-player.ts [hunk 2]
```diff
@@ -10,34 +10,50 @@ export {};
...
+		"facebook",
+		"twitter",
+	];
+
+	return !filteredTerms.some(
+		(term) => artist?.includes(term) || track?.includes(term)
+	);
 
-	return false;
+	return !filteredTerms.track?.includes("@");
```
Comment (inverse) [inline src/connectors/siriusxm-player.ts:52]: Should this be: 

```suggestion
	return !track.includes("@");
```

Comment (jbwharris) [reply src/connectors/siriusxm-player.ts]: @inverse I just removed that line. Was experimenting with another idea for the "@" symbol, but found it was too greedy.

**Pull request merged**